orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -33,15 +33,11 @@
 @Component
 public class TrafficGuard {
 
-  private final OortHelper oortHelper;
-
-  private final Front50Service front50Service;
-
   @Autowired
-  public TrafficGuard(OortHelper oortHelper, Front50Service front50Service) {
-    this.oortHelper = oortHelper;
-    this.front50Service = front50Service;
-  }
+  private OortHelper oortHelper;
+
+  @Autowired(required = false)
+  private Front50Service front50Service;
```

Comment:
I have to say I vastly prefer constructor injection. I know Spinnaker generally isn't consistent on that point, though. It's possible to make constructor injected dependencies optional by declaring them as e.g. `Optional<Front50Service>`.